### ISSUE TITLE
Creating the objects for the envelope condition (loss function)  programmatically from a Block using autograd

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#106]
In order to solve the optimization problem embedded in the Bellman version of the Maliar training loop, there are three techniques (from MMW '21)

<img width="1272" height="186" alt="Image" src="https://github.com/user-attachments/assets/86e90ca3-5bba-4c06-8bb7-751a634f9829" />

We would like to be able to get these kinds of objects programmatically from a Block.

Starting with the Envelope condition (which is simplest):

- see if the `reward_function` for the block can be autodifferentiated using `torch` to get the $r_s(m,s,x)$ values
- likewise for the value function (currently available from the BlockValueNet)
- combine these into a loss function appropriately. (maybe automate this step also)
- connect it to the Maliar training loop (see #105)
